lib/src/package_meta.dart

```diff
@@ -360,13 +329,10 @@ class _FilePackageMeta extends PubPackageMeta {
   /// empty list if no reasons found.
   @override
   List<String> getInvalidReasons() {
-    var reasons = <String>[];
-    if (_pubspec.isEmpty) {
-      reasons.add('no pubspec.yaml found');
-    } else if (!_pubspec.containsKey('name')) {
-      reasons.add('no name found in pubspec.yaml');
-    }
-    return reasons;
+    return [
```

Comment:
It feels a little weird how this List and its conditional values are regenerated every time we want to check `isValid`. If `_pubspec` is final, could we not have just saved this as a field or something?

Reply:
It should be easy to compute this value, even if multiple times (I think it's only a few times per package; not like a few times per field-being-documented or something.

We're following ["Avoid storing what you can calculate"](https://dart.dev/effective-dart/usage#avoid-storing-what-you-can-calculate) here. (Not that it would take much storage, one field per package as well.)

What I'd like to do is rename it and make it a getter (so it could be a field (without creating a private backing field)), but that is a breaking change with google3 so... too much work.

Reply:
Fair enough. Thanks for the explanation!